shared/utils/env_utils.py

```diff
@@ -171,3 +172,77 @@ def pawns_around(board: Board, pawn: tuple, distance: int):
             if (x+i, y+j) in board.get_black_coordinates():
                 count += 1
     return count
+
+def piece_parser(piece: Piece) -> int:
+    """
+    Return the index of the boolean array (which represents the board) used as a input for the policy network of the DQN
+
+    Arg:
+    Piece object
+
+    Example:
+    If the piece given is the KING, the function will return 1
+    The second array given as input will be the one displaying the position of the KING in the 9x9 board (index 1 means second element)
+    """
+    state_pieces = {Piece.DEFENDER : 0,
+                    Piece.KING : 1,
+                    Piece.ATTACKER : 2,
+                    Piece.EMPTY : 3,
+                    Piece.CAMPS : 4,
+                    Piece.THRONE : 4}
+    return state_pieces[piece]
+
+class FeaturizeState:
+    """
+    Class representing the state given as input to the DQN.
+
+    Methods:
+        generate_input(): Generates the tensor input of the DQN from the position of the pieces, the turn and the points given from the black and white heuristics
+    """
+    def __init__(self, state_string: str):
+        """
+        Initialize the State Featurizer class with the string representing the board
+        This string is used to generate the tensor input for the DQN
+        
+        Arg:
+        state_string, the string representing the board
+        """
+        self.state_string = state_string
+
+    def generate_input(self):
```

Comment:
declare a single static method which takes a State as input, instead of the __init__